File: C_Programming/Study/Q5_Assignment28.c
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name :  Pattern
// Description:     Assignment_28: Q.5 :Accept a number of rows & number of columns from user & Display below pattern.                                       
// Input : iRow = 5          iCol = 5
// Output:   1    2    3    4    5     
//           1    2              5
//           1         3         5
//           1              4    5 
//           1    2    3    4    5         
//          
// Date:            15/06/2026
// Author:          Shital Ajit Nikam   
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<stdio.h>

void Pattern(int iRow, int iCol)
{
        int i = 0, j = 0; 
                                                    
        for(i = 1; i <= iRow; i++)
        {                 
            for (j = 1; j <= iCol; j++)
            {     
                if(i == 1 || i == iCol || j == 1 || j == iRow)
                {                                          
                    printf("%d\t",j);   
                }
                else if(i == j)
                {
                   printf("%d\t",i); 
                }
                else
                {
                    printf(" \t");
                }
            }                            
            printf("\n");                     
        }                         
    
}

int main()
{       
    int iValue1 = 0, iValue2;

    printf("Enter Number of Rows \n");
    scanf("%d",&iValue1);
               
    printf("Enter Number of Column \n");
    scanf("%d",&iValue2);
        
    Pattern(iValue1, iValue2);   
        
    return 0;                           
        
}

File: Python_Programming/Competative/Q5_Assignment28.py
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
#        
# Description: Assignment_28: Q.5: write a program which accept file name & a word from user, and check whether 
#                                  that word present in the file or not.
# Input  :  Demo.txt Marvellous
# Output :  Display whether word found in the file or not.                                    
#                                     
# Date:            19/07/2026
# Author:          Shital Ajit Nikam   
#
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import os

#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
# Function Name: SearchWord: Search the word in the file.
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////


def SearchWord(FileName, word):
    try:
        fobj1 = open(FileName,"r")                  # read mode
        data = fobj1.read()
        
        if word in data:
            return True
        else:
            return False
      
    
    except FileNotFoundError :
        print("Unable to open file, file does not exist.")
        return None            
                

def main():        
        
        Border = "-"*60
        print(Border)         
        
        fname = input("Enter the Source File name :")
        fword = input("Enter the word to search :")
        
        print(Border)
               
        Ret = SearchWord(fname,fword)    
        if (Ret == True) :
            print("Word found in the file.")
        else :
            print("Word not found in the file.")                                   
                   
        print(Border)                    
    
        
if __name__ == "__main__" :
    main()